xclim/core/units.py

```diff
@@ -302,6 +314,110 @@ def convert_units_to(
     raise NotImplementedError(f"Source of type `{type(source)}` is not supported.")
 
 
+FREQ_UNITS = {
+    "N": "nanosecond",
+    "L": "millisecond",
+    "S": "second",
+    "T": "minute",
+    "H": "hour",
+    "D": "day",
+    "W": "week",
+    "M": "month",
+    "A": "year",
+}
+
+
+def infer_sampling_units(da: xr.DataArray, deffreq: str = "D") -> Tuple[int, str]:
+    """Infer a multiplicator and the units corresponding to one sampling period.
+
+    If `xr.infer_freq` fails, returns defval
+    """
+    freq = xr.infer_freq(da.time)
+    if freq is None:
+        freq = deffreq
+
+    multi, base, _ = parse_offset(freq)
+    try:
+        return int(multi or "1"), FREQ_UNITS[base]
+    except KeyError:
+        raise ValueError(f"Sampling frequency {freq} has no corresponding units.")
+
+
+def to_agg_units(
+    out: xr.DataArray, orig: xr.DataArray, op: str, dim="time"
+) -> xr.DataArray:
+    """Set and convert units of an array after an aggregation operation along the sampling dimension (time).
```

Comment:
Is seems it also changes the data itself no ? (out = out * m)

This function should have a dedicated test. 
Also please include a conceptual example of what it does.